src/java/org/apache/cassandra/db/ReadVerbHandler.java

```diff
@@ -49,6 +50,8 @@ public void doVerb(MessageIn<ReadCommand> message, int id)
                                                                       ReadResponse.serializer);
         Tracing.trace("Enqueuing response to {}", message.from);
         MessagingService.instance().sendReply(reply, id, message.from);
+
+        OwnershipVerificationUtils.verifyRead(keyspace, command.key);
     }
```

Comment:
note that this verification is intentionally after enqueuing the reply in order to avoid introducing additional latency on the read path. We should consider moving it earlier and throwing an exception after understanding in what circumstances this violation is logged, if any.